plugin/src/test/groovy/com/novoda/buildproperties/BuildPropertiesTest.groovy

```diff
@@ -90,35 +90,37 @@ class BuildPropertiesTest {
     void shouldThrowWhenAccessingPropertyFromNonExistentPropertiesFile() {
         project.buildProperties {
             foo {
-                file project.file('foo.properties')
+                from project.file('foo.properties')
             }
         }
 
         try {
             project.buildProperties.foo['any'].string
-            fail('Gradle exception not thrown')
-        } catch (GradleException e) {
-            assertThat(e.getMessage()).endsWith('foo.properties does not exist.')
+            fail('Exception not thrown')
+        } catch (Exception e) {
+            assertThat(e.getMessage()).contains('foo.properties does not exist.')
         }
     }
 
     @Test
     void shouldProvideSpecifiedErrorMessageWhenAccessingPropertyFromNonExistentPropertiesFile() {
         project.apply plugin: BuildPropertiesPlugin
 
-        def errorMessage = 'This file should contain the following properties:\n- foo\n- bar'
+        def description = 'This file should contain the following properties:\n- foo\n- bar'
+        def consoleRenderer = new ConsoleRenderer()
         try {
             project.buildProperties {
                 foo {
-                    file project.file('foo.properties'), errorMessage
+                    from project.file('foo.properties')
+                    setDescription(description)
                 }
             }
             project.buildProperties.foo['any'].string
-            fail('Gradle exception not thrown')
-        } catch (GradleException e) {
+            fail('Exception not thrown')
+        } catch (Exception e) {
             String message = e.getMessage()
             assertThat(message).contains('foo.properties does not exist.')
-            assertThat(message).endsWith(errorMessage)
+            assertThat(message).contains(consoleRenderer.indent(description, "* buildProperties.foo: "))
```

Comment:
Is it a good idea to have your test assertions to be dependent on the `consoleRenderer`?